### PR TITLE
fix(color-interpolate): add color-interpolate file

### PR DESCRIPTION
**Pull request by llekn**

### CODE REVIEW

File: app/assets/javascripts/color-interpolate.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
+
+		if (result[3] === 1) {
+			return 'rgb(' + result.slice(0, 3) + ')';
+		}
+		return 'rgba(' + result + ')';
+	};
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 0 spaces but found 1 tab                 indent

File: app/assets/javascripts/color-interpolate.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
+
+		if (result[3] === 1) {
+			return 'rgb(' + result.slice(0, 3) + ')';
+		}
+		return 'rgba(' + result + ')';
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 2 spaces but found 2 tabs                indent<br>Expected newline before return statement                         newline-before-return<br>Unexpected string concatenation                                  prefer-template

File: app/assets/javascripts/color-interpolate.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
+
+		if (result[3] === 1) {
+			return 'rgb(' + result.slice(0, 3) + ')';
+		}
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 0 spaces but found 2 tabs                indent

File: app/assets/javascripts/color-interpolate.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
+
+		if (result[3] === 1) {
+			return 'rgb(' + result.slice(0, 3) + ')';
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 2 spaces but found 3 tabs                indent<br>Unexpected string concatenation                                  prefer-template<br>No magic number: 0                                               no-magic-numbers<br>No magic number: 3                                               no-magic-numbers

File: app/assets/javascripts/color-interpolate.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
+
+		if (result[3] === 1) {
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 2 spaces but found 2 tabs                indent<br>No magic number: 3                                               no-magic-numbers<br>No magic number: 1                                               no-magic-numbers

File: app/assets/javascripts/color-interpolate.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		t = idx - lIdx;
+
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
+		});
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 0 spaces but found 2 tabs                indent

File: app/assets/javascripts/color-interpolate.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+		t = idx - lIdx;
+
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
+			return v;
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected newline before return statement                         newline-before-return<br>Expected indentation of 2 spaces but found 3 tabs                indent

File: app/assets/javascripts/color-interpolate.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+		    rIdx = Math.ceil(idx);
+
+		t = idx - lIdx;
+
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
+			if (i < 3) v = Math.round(v);
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 2 spaces but found 3 tabs                indent<br>No magic number: 3                                               no-magic-numbers<br>Assignment to function parameter 'v'                             no-param-reassign

File: app/assets/javascripts/color-interpolate.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		    lIdx = Math.floor(idx),
+		    rIdx = Math.ceil(idx);
+
+		t = idx - lIdx;
+
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
+			v = mix(v, rColor[i], t);
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: Expected indentation of 2 spaces but found 3 tabs                indent<br>Assignment to function parameter 'v'                             no-param-reassign

File: app/assets/javascripts/color-interpolate.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+		var idx = (palette.length - 1) * t,
+		    lIdx = Math.floor(idx),
+		    rIdx = Math.ceil(idx);
+
+		t = idx - lIdx;
+
+		var lColor = palette[lIdx],
+		    rColor = palette[rIdx];
+
+		var result = lColor.map(function (v, i) {
```
Comment (monkeyci) [inline app/assets/javascripts/color-interpolate.js]: All 'var' declarations must be at the top of the function scope  vars-on-top<br>Unexpected var, use let or const instead                         no-var<br>Expected indentation of 2 spaces but found 2 tabs                indent<br>Unexpected function expression                                   prefer-arrow-callback

**Comment by llekn** [comment]
Perdón, se me pasó subir el archivo en #40 .

El archivo no se ajusta para nada a los estándares platanescos, pero en honor al tiempo y a no romper nada, se deshabilitó el linter.

**Pull request merged**